src/modules/facility/facility.service.ts

```diff
@@ -51,4 +56,213 @@ export class FacilityService {
       probabilityOfDefault: facilityInAcbs.ProbabilityofDefault,
     };
   }
+
+  async createFacility(facilityToCreate: CreateFacilityRequestItem): Promise<void> {
+    const { portfolioIdentifier } = PROPERTIES.GLOBAL;
+    const idToken = await this.acbsAuthenticationService.getIdToken();
+    const newFacilityInAcbs = this.buildAcbsCreateFacilityRequest(facilityToCreate);
+    await this.acbsFacilityService.createFacility(portfolioIdentifier, newFacilityInAcbs, idToken);
+  }
+
+  private buildAcbsCreateFacilityRequest(facilityToCreate: CreateFacilityRequestItem): AcbsCreateFacilityRequest {
+    const defaultValues = PROPERTIES.FACILITY.DEFAULT.POST;
+
+    const { compBalPctReserve, userDefinedDate1, isUserDefinedDate1Zero } = this.buildFacilityStageDerivedValuesToCreate(facilityToCreate);
+
+    const acbsGuaranteeExpiryDate = this.dateStringTransformations.addTimeToDateOnlyString(facilityToCreate.guaranteeExpiryDate);
+    const midnightToday = this.dateStringTransformations.getDateStringFromDate(new Date());
+
+    const capitalConversionFactorCode =
+      facilityToCreate.capitalConversionFactorCode ??
+      defaultValues.capitalConversionFactorCode[facilityToCreate.productTypeId] ??
+      defaultValues.capitalConversionFactorCodeFallback;
+
+    const description = this.buildFacilityDescriptionToCreate(facilityToCreate);
+    const effectiveDateString = this.getFacilityEffectiveDateToCreate(facilityToCreate);
+
+    return {
+      FacilityIdentifier: facilityToCreate.facilityIdentifier,
+      Description: description,
+      Currency: {
+        CurrencyCode: facilityToCreate.currency,
+        IsActiveIndicator: true,
+      },
+      OriginalEffectiveDate: effectiveDateString,
+      DealIdentifier: facilityToCreate.dealIdentifier,
+      DealPortfolioIdentifier: facilityToCreate.portfolioIdentifier ?? PROPERTIES.GLOBAL.portfolioIdentifier,
+      DealBorrowerPartyIdentifier: facilityToCreate.dealBorrowerIdentifier,
+      BookingDate: midnightToday, // TODO APIM-69: should actually be now(), but docs say it is ignored anyway
```

Comment:
@avaitonis do you have any concerns with this? Mulesoft uses the current date time (including the time part) but I've changed it to be all 0s for the time part. The ACBS documentation says this field is ignored on POST requests anyway.

Reply:
Yes, midnightToday looks good.

You could try to remove BookingDate and see if works. Couple years ago ACBS was requiring some of these "Not required" fields. Safer and my preferred option would be to keep sending it in POST.

Reply:
Thanks Audrius, I will keep sending it. I agree it is safer